Upgrade to TypeSript 3.6

Looks like we'll have some type adjustments to make in the typegen module.

IIRC this is b/c TypeScript introduced changes to how to handles tuples.

```
❯ yarn test
yarn run v1.19.0
$ jest
 FAIL  tests/backingTypes.spec.ts
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/typegen.ts:482:13 - error TS2322: Type 'Record<string, string[]>' is not assignable to type 'Record<string, [string, string]>'.
      Index signatures are incompatible.
        Type 'string[]' is not assignable to type '[string, string]'.

    482             argTypeMap[type.name][field.name] = field.args.reduce(
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

 FAIL  tests/sdlConverter.spec.ts
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/typegen.ts:482:13 - error TS2322: Type 'Record<string, string[]>' is not assignable to type 'Record<string, [string, string]>'.
      Index signatures are incompatible.
        Type 'string[]' is not assignable to type '[string, string]'.

    482             argTypeMap[type.name][field.name] = field.args.reduce(
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/typegen.ts
+++ b/src/typegen.ts
@@ -480,7 +480,7 @@ export class Typegen {
           if (field.args.length > 0) {
             argTypeMap[type.name] = argTypeMap[type.name] || {};
             argTypeMap[type.name][field.name] = field.args.reduce(
-              (obj: Record<string, string[]>, arg) => {
+              (obj: Record<string, [string, string]>, arg) => {
                 obj[arg.name] = this.normalizeArg(arg);
                 return obj;
               },

